server/sonar-server/src/main/java/org/sonar/server/db/migrations/MassUpdate.java

```diff
@@ -76,8 +80,12 @@ public void execute(final Handler handler) throws SQLException {
       select.scroll(new Select.RowHandler() {
         @Override
         public void handle(Select.Row row) throws SQLException {
-          if (handler.handle(row, update)) {
-            update.addBatch();
+          try {
+            if (handler.handle(row, update)) {
+              update.addBatch();
+            }
+          } catch (Exception e) {
+            throw new IllegalStateException("Error during processing of row: [" + row + "]", e);
           }
```

Comment:
This try catch should be moved in each method of SelectImpl (list, get and scroll around reader.read(row) and handler.handle(row)), in order to manage more exceptions.